xpra/keyboard/layouts.py

```diff
@@ -296,7 +296,7 @@
     0x00000463  : ("af", "Pashto (Afghanistan)"),
     #duplicate of 'ku'
     #0x00000429  : ("ir", "Persian"),
-    0x00050429  : ("ir", "Persian (Standard)"),
+    0xa0000429  : ("ir", "Persian (Standard)"),
```

Comment:
This conflicts with:
```
    0x00000429  : ("ku", "Central Kurdish"),
```
Which is why it is commented out in the line just above - I have no idea which one is correct.
If it is indeed this one, then the other one should be commented out.

Reply:
For the Persian language, the 0xa0000429 is the correct one in Windows. Before introduction of the "Persian (Standard)" layout, the "ku" layout was used as a workaround, but since Windows 10, the default is changed (and matches with the Linux choice thereafter).

```python
            for hkl in l:                                               
                for mask, bitshifts in KMASKS.items():                  
                    kbid = 0                                            
                    for bitshift in bitshifts:                          
                        kbid = (hkl & mask)>>bitshift                   
                        if kbid in WIN32_LAYOUTS:                       
                            break                                       
                    if kbid in WIN32_LAYOUTS:                           
                        code, _, _, _, _layout, _variants = WIN32_LAYOUTS.get(kbid)
                        log("found keyboard layout '%s' / %#x with variants=%s, code '%s' for kbid=%#x",                
                            _layout, kbid, _variants, code, hkl)        
                        if _layout not in layouts_defs:                 
                            layouts_defs[_layout] = hkl                 
                            break
```

Shouldn't we keep both uncommented since they are describing different layout names? ir vs ku?
Users either have **ku** (before Windows 10) or **ir** (after Windows 10) and if a user enable both layouts manually, since the iteration on `KMASKS.items()` is ordered, the ku=0x00000429 will not catch the ir=0xa0000429 layout.
That is, an `hkl` which is supposed to match with **ir**, will match with it in the first iteration (and in the second iteration, `if _layout not in layouts_defs:` prevents the **ir** detected layout to be overridden) while an `hkl` which is going to match with **ku** will never match with the **ir** (due to its extra 0x0a byte).

Reply:
You're absolutely right, I had missed the `0xa` prefix.